app/build.gradle.kts

```diff
@@ -0,0 +1,315 @@
+    implementation(libs.androidx.preference)
+    implementation(libs.androidx.constraintlayout)
+    implementation(libs.rhymer)
+    implementation(libs.porter.stemmer)
+    implementation(libs.dagger)
+    implementation(libs.androidx.room.runtime)
+    implementation(libs.kotlin)
+    implementation(libs.kotlinx.coroutines.core)
+    implementation(libs.kotlinx.coroutines.android)
+
+    // We need to explicitly add a couple of api dependencies here, otherwise alpha versions
+    // of these libs will be pulled in transitively (by a non-alpha databinding dependency...)
+    api(libs.androidx.lifecycle.runtime)
+    api(libs.androidx.lifecycle.viewmodel.ktx)
+    api(libs.androidx.collection)
+
+    ksp(libs.androidx.room.compiler)
+    ksp(libs.dagger.compiler)
+
+    androidTestImplementation(libs.androidx.room.testing)
+
+    testImplementation(libs.junit)
+    testImplementation(libs.androidx.test.ext)
+    testImplementation(libs.robolectric)
+    testImplementation(libs.androidx.test.runner)
+    testImplementation(libs.androidx.test.rules)
+    testImplementation(libs.androidx.test.espresso.core)
+    testImplementation(libs.androidx.test.espresso.contrib)
+    testImplementation(libs.androidx.test.espresso.intents)
+    testImplementation(libs.fest.reflect)
+    testImplementation(libs.androidx.arch.core.testing)
+    testImplementation(libs.kotlinx.coroutines.test)
+    kspTest(libs.dagger.compiler)
+
+    androidTestImplementation(libs.fest.reflect)
+    androidTestImplementation(libs.androidx.test.runner)
+    androidTestImplementation(libs.androidx.test.rules)
+    androidTestImplementation(libs.androidx.test.espresso.core)
+    androidTestImplementation(libs.androidx.test.espresso.contrib)
+    androidTestImplementation(libs.androidx.test.espresso.intents)
+    androidTestImplementation(libs.androidx.test.core)
+    androidTestImplementation(libs.androidx.test.ext)
+    androidTestImplementation(libs.androidx.test.uiautomator)
+    androidTestImplementation(libs.robolectric.annotations)
+    androidTestImplementation(libs.google.test.parameter.injector)
+
+    androidTestUtil(libs.androidx.test.orchesetrator)
+}
+
+// Only show real releases with the ben-manes plugin.
+tasks.named<DependencyUpdatesTask>("dependencyUpdates").configure {
+    resolutionStrategy {
+        componentSelection {
+            all { selection: ComponentSelection ->
+                var rejected = listOf("alpha", "alpha-preview", "beta", "rc", "cr", "m", "eap", "dev").any { qualifier ->
+                    selection.candidate.version.matches(Regex("(?i).*[.-]${qualifier}[.\\d-]*"))
+                }
+                if ("com.android.databinding" == selection.candidate.group) {
+                    rejected = true
+                }
+                if (rejected) {
+                    selection.reject("Release candidate")
+                }
+            }
+        }
+    }
+}
+tasks.register<JacocoReport>("jacocoTestReport") {
```

Comment:
Something may be incorrect with the jacocoTestReport.

It may not be merging the unit test and instrumentation test reports.

The new report shows `importFavorites` as not covered. This is only tested by unit tests.